core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1643,8 +1643,18 @@ private Scope createScopeForView(
         private List<Field> analyzeTableOutputFields(Table table, QualifiedObjectName tableName, TableSchema tableSchema, Map<String, ColumnHandle> columnHandles)
         {
             // TODO: discover columns lazily based on where they are needed (to support connectors that can't enumerate all tables)
+            QualifiedObjectName name = createQualifiedObjectName(session, table, table.getName());
+            Set<String> accessibleColumnNames = accessControl.filterColumns(
+                    session.toSecurityContext(),
+                    name.asCatalogSchemaTableName(),
+                    tableSchema.getColumns().stream()
+                            .map(ColumnSchema::getName)
+                            .collect(toImmutableSet()));
+            List<ColumnSchema> accessibleColumns = tableSchema.getColumns().stream()
+                    .filter(column -> accessibleColumnNames.contains(column.getName()))
```

Comment:
This behavior is non standard (non-specwise) AFAIK, so should be an opt-in behavior, guarded with a toggle.

Reply:
Where should the toggle knob be located, in your opinion? I imagine it could be in the SystemAccessControl implementation itself if we pass a flag when calling to tell the AccessControl whether this call is for SELECT or for metadata. 

Or, should there be another method in the SystemAccessControl interface like 
`boolean shouldFilterColumns(TableName name)` that we check first? This would give the AccessControl implementations the ability to behave differently for different tables. 

I suppose the more fine grained control we give to the AccessControl implementations, the closer this becomes to the implementation in #7893 where we expose two methods and let the SystemAccessControl decide entirely when it wants to hide/throw on inaccessible columns. 

I think if we _really_ want to maintain a consistent behavior here, we should also take a look at `checkCanSelectFromColumns`, and maybe enforce that if a column is supposed to be inaccessible then the toggle only controls whether the behavior is to throw or hide.

Reply:
I am not sure if toggle belongs to access control. To me it belongs to the engine as it we change the semantic of SQL. Authorization works same way, but its result is interpreted differently.

Reply:
Hmmm - is there any value in allowing access control implementations to decide which tables should throw vs which tables should hide? If not, then toggling in the engine makes sense to me.

@findepi what do you think?

Reply:
I don't see any value in making this behave differently for a subset of tables.